### PR TITLE
Store auto prompt impressions/dismissals to local storage, and use it to determine whether to show subsequent prompts according to the AutoPromptConfig.

### PR DESCRIPTION
**Pull request by stellachui**

### CODE REVIEW

File: src/runtime/client-config-manager.js [hunk 1]
```diff
@@ -129,7 +129,19 @@ export class ClientConfigManager {
     let autoPromptConfig = undefined;
     if (autoPromptConfigJson) {
       autoPromptConfig = new AutoPromptConfig(
-        autoPromptConfigJson['maxImpressionsPerWeek']
+        autoPromptConfigJson['maxImpressionsPerWeek'],
+        autoPromptConfigJson['clientDisplayTrigger'] &&
+          autoPromptConfigJson['clientDisplayTrigger']['dismissalDelaySeconds'],
```
Comment (ChrisAntaki) [inline src/runtime/client-config-manager.js:134]: Could you use the optional chaining operator (?.) here to simplify the code a bit?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

File: src/model/auto-prompt-config.js [hunk 2]
```diff
@@ -19,10 +19,64 @@
...
+    this.explicitDismissalConfig = new ExplicitDismissalConfig(
+      backoffSeconds,
+      maxDismissalsPerWeek,
+      maxDismissalsResultingHideSeconds
+    );
+  }
+}
+
+/**
+ *
```
Comment (ChrisAntaki) [inline src/model/auto-prompt-config.js:47]: Could you add a description for this class?

File: src/model/auto-prompt-config.js [hunk 3]
```diff
@@ -19,10 +19,64 @@
...
+   * @param {number|undefined} dismissalDelaySeconds
+   */
+  constructor(dismissalDelaySeconds) {
+    /** @const {number|undefined} */
+    this.dismissalDelaySeconds = dismissalDelaySeconds;
+  }
+}
+
+/**
+ *
```
Comment (ChrisAntaki) [inline src/model/auto-prompt-config.js:60]: Could you add a description for this class?

File: src/runtime/client-config-manager.js (hunk 1 above)
Comment (ChrisAntaki) [reply src/runtime/client-config-manager.js:134]: Sorry, nevermind, looks like our Closure JS compiler doesn't support this

**Review by ChrisAntaki — APPROVED** [review]

File: src/model/auto-prompt-config.js (hunk 2 above)
Comment (stellachui) [reply src/model/auto-prompt-config.js]: Done.

File: src/model/auto-prompt-config.js (hunk 3 above)
Comment (stellachui) [reply src/model/auto-prompt-config.js]: Done.

**Pull request merged**